### PR TITLE
Gp/support indexing for p2pk

### PR DESCRIPTION
**Pull request by JackPiri**
This PR adds support for Pay-To-Public-Key (P2PK) scripts; before this PR only Pay-To-Public-Key-Hash (P2PKH) and Pay-To-Script-Hash (P2SH) scripts were supported.
PR associated to issue [#352](https://github.com/HorizenOfficial/zen/issues/352).
Notice: this is a rebased clone of PR [#480](https://github.com/HorizenOfficial/zen/pull/480).

### CODE REVIEW

**Review by drgora — APPROVED** [review]
Looks good to me

File: src/main.h [hunk 1]
```diff
@@ -46,6 +46,9 @@ class CValidationState;
 class CTxUndo;
 struct CNodeStateStats;
 class CTxInUndo;
+#ifdef ENABLE_ADDRESS_INDEXING
```
Comment (a-petrini) [inline src/main.h]: This define has been removed from the codebase with PR https://github.com/HorizenOfficial/zen/pull/531 so it should not be introduced back.

File: src/rpc/blockchain.cpp [hunk 2]
```diff
@@ -3,7 +3,9 @@
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
+#ifdef ENABLE_ADDRESS_INDEXING
```
Comment (a-petrini) [inline src/rpc/blockchain.cpp]: As in `main.h`, this define has been removed from the codebase with PR https://github.com/HorizenOfficial/zen/pull/531 so it should not be introduced back.

File: src/rpc/misc.cpp [hunk 3]
```diff
@@ -3,7 +3,9 @@
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
+#ifdef ENABLE_ADDRESS_INDEXING
```
Comment (a-petrini) [inline src/rpc/misc.cpp]: As in `main.h`, this define has been removed from the codebase with PR https://github.com/HorizenOfficial/zen/pull/531 so it should not be introduced back.

**Review by a-petrini — CHANGES_REQUESTED** [review]

File: src/rpc/blockchain.cpp (hunk 2 above)
Comment (JackPiri) [reply src/rpc/blockchain.cpp]: Yes, this is a wrong rebasing; modified accordingly.

File: src/main.h (hunk 1 above)
Comment (JackPiri) [reply src/main.h]: Yes, this is a wrong rebasing; modified accordingly.

File: src/rpc/misc.cpp (hunk 3 above)
Comment (JackPiri) [reply src/rpc/misc.cpp]: Yes, this is a wrong rebasing; modified accordingly.

File: src/main.h (hunk 1 above)
Comment (a-petrini) [reply src/main.h]: Nice, thanks!

**Review by a-petrini — APPROVED** [review]

**Review by ptagl — APPROVED** [review]

**Pull request merged**